src/store/index.ts

```diff
@@ -54,41 +64,73 @@ export const resultsFilter = ref<ResultsFilter>({
   },
 });
 
-export const addFootprint = (spatialId: number) => {
-  if (footprintIds.value === undefined) {
-    footprintIds.value = [];
+export const addFootprint = async (spatialId: number, region?: boolean) => {
+  let key;
+  let footprint;
+  if (!region) {
+    footprint = (await rgdFootprint(spatialId)).footprint;
+    key = `result_${spatialId}`;
+  } else {
+    footprint = regionsList.value?.find((reg) => reg.id === spatialId)?.footprint;
+    key = `region_${spatialId}`;
+  }
+  if (key && footprint) {
+    visibleFootprints.value = { [key]: footprint, ...visibleFootprints.value };
   }
-  footprintIds.value.push(spatialId);
 };
 
-export const removeFootprint = (spatialId: number) => {
-  footprintIds.value = footprintIds.value.filter((obj: number) => obj !== spatialId);
+export const removeFootprint = (spatialId: number, region?: boolean) => {
+  let key: string;
+  if (!region) {
+    key = `result_${spatialId}`;
+  } else {
+    key = `region_${spatialId}`;
+  }
+  if (visibleFootprints.value[key]) {
+    visibleFootprints.value = Object.fromEntries(
+      Object.entries(visibleFootprints.value).filter(([k]) => k !== key),
+    );
```

Comment:
Why not just `delete visibleFootprints.value[key]`?

Reply:
For some reason, this doesn't work with `ref` objects... 🤷🏻‍♂️